File: c++/TopSongs.cpp
/*
Sample code to read the contents of a file containing information delimited by '||' and print out the information

Command Line Instructions to Build and Run (if it's been a while since you've used the command line):
g++ -o TopSongs TopSongs.cpp
./TopSongs
*/

#include <iostream>
#include <fstream>
#include <string>

using namespace std;

int main()
{
    ifstream fileReader;
    fileReader.open("../topSongs.txt");//open the file and process it - let's just pretend there's error checking

    if(fileReader)
    {
        string line, token;
        string delimiter = "||";
        int pos;
        while(getline(fileReader, line))//loop through the file and read it line by line
        {
            //I wish there was a standard implementation of a split function to use!!!
            pos = 0;
            while ((pos = line.find(delimiter)) != string::npos)
            {
                token = line.substr(0, pos);
                cout<<token<<endl;
                line.erase(0, pos + delimiter.length());
            }
            cout<<line<<endl
                      <<endl;
        }
        fileReader.close();//we're done!
    }
    else
    {
        cout<<"We've got a problem!"<<endl;
    }
    return 0;
}

File: java/TopSongs.java
/*
Sample code to read the contents of a file containing information delimited by '||' and print out the information

Command Line Instructions to Build and Run (if it's been a while since you've used the command line):
javac TopSongs.java
java TopSongs
*/

import java.util.Scanner;
import java.io.File;
import java.io.IOException;

public class TopSongs
{
   public static void main(String[] args)
   {
       try
       {
            Scanner songFile = new Scanner(new File("../topSongs.txt"));//open the file for reading via a Scanner

            while(songFile.hasNext())
            {
                String[] tokens = songFile.nextLine().split("\\|\\|");//split each line into tokens on "||""

                for(String token : tokens)
                {
                    System.out.println(token);//print each token on it's own line
                }
                System.out.println();//print a blank line between records in the file
            }

            songFile.close();//we're done!
       }
       catch(IOException e)
       {
           System.out.println("Error: " + e.getMessage());
       }
   }
}